File: src/main/java/br/com/adamastor/votacao/core/dominio/modelo/Voto.java
package br.com.adamastor.votacao.core.dominio.modelo;

import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;

import java.time.Instant;
import java.util.UUID;

@Getter
@Builder
public class Voto {

    @EqualsAndHashCode.Include
    private final UUID id;

    private final UUID sessaoId;

    private final String cpfAssociado;

    private final VotoOpcao opcao;

    private final Instant dataHoraCriacao;

}

File: frontend/src/types/voto.ts
export type OpcaoVoto = 'SIM' | 'NAO';

export interface RegistrarVotoRequisicaoDTO {
  cpfAssociado: string;
  opcao: OpcaoVoto;
}

export interface VotoRespostaDTO {
  id: string;
  sessaoId: string;
  cpfAssociado: string;
  opcao: OpcaoVoto;
  dataHoraCriacao: string;
}
